tests/CMakeLists.txt

```diff
@@ -1,3 +1,31 @@
+## Function to set test labels
+## Test labels should be added for the following categories (USE ALL CAPS)
+#- Test Group
+#	- NNLS, TGV_SCALING, ETC
+#- Dimension
+#   - 1D, 2D, 3D
+#- Parallel vs Serial
+#   - PARALLEL, SERIAL
+#- PDE Type
+#   - EULER, NAVIER-STOKES, etc
+#- ODE Solver Type
+#   - RUNGE-KUTTA, IMPLICIT, etc
+#- DG Type
+#   - STRONG, STRONG-SPLIT, WEAK
+#- Quadrature Type
+#   - COLLOCATED, UNCOLLOCATED
+#- OTHER (if needed)
+#	- MEMORY_INTENSIVE, MANUFACTURED_SOLUTION, EXPECTED_FAILURE, CONVERGENCE etc
```

Comment:
Could you add all of the "OTHER" possibilities here?

Also, my preference would be to have a separate readme/markdown file with this information, but this is also good. As long as it's clear to everyone how to add consistent label names to their tests.

Reply:
IMO, the description you wrote for this PR would be good as a read-me. Can others please comment if they have opinions on how to make labelling conventions easy to follow?

Reply:
+1 for including how to add and use labels to readme.